### PR TITLE
add FreeForm hours conversion

### PR DESCRIPTION
**Pull request by bill-paren**
### Add free-form opening hours parsing
**Summary**
Adds ParseFreeformOpeningHours / FreeformOpeningHoursToISOString, which convert human-written opening-hours strings (as commonly seen in POI/EV-charging datasets) into []OpeningHours / the package's ISO-like string format. 

This code was built incrementally with Claude by feeding him over 400 various formats from the AFDC AccessDaysTime data column

**Supported formats**
Every-day shorthand: "9am-9pm", "5am to 10pm daily", "24 hours daily", "24/7"
Day lists and ranges: "9am-9pm M-Th, 9am-6pm F, 9am-5pm Sat", with abbreviations from single-letter (M, F) through three-letter (Mon, Sat) to full names, plus "weekdays"/"every day" keywords
Compound day specs: joined by comma, "and", or "&" — "W-F and Sun", "Mon, Wed and Fri", "M & W"
Overnight/cross-midnight ranges: "10pm-6am" correctly rolls the closing day forward
Multiple time windows sharing one day spec: "12am-7am, 9am-5:30pm, 7:30pm-12am daily", "8am-12pm and 1pm-5pm M-F", "6:30am-6pm & 8pm-10pm daily"
"24 hours <day spec>" as a standalone clause: "3pm-7:45am M-F, 24 hours Sat-Sun"
Free-text notes are dropped, not fatal: semicolon-delimited ("...; for staff use only"), parenthetical ("(May-Sept)"), or glued on with no separator at all ("M-F daily", "daily for guests", "Sat, for client use only") — the parser recovers the longest valid leading day-spec and silently drops the rest
"closed <day>" as an explicit (no-op) exclusion clause
Minor normalizations: 12am as a closing time means end-of-day (24:00), not start-of-day; a bare T means Tuesday (this abbreviation style always spells out Th for Thursday); S-S means Saturday-Sunday; a missing opening-time meridiem defaults to AM ("9-5pm")

**Design notes**
Each format is a small, independently-testable parser function registered in an ordered list (freeformParsers for whole-string formats, segmentParsers for individual clauses within a multi-day string), so new formats can be added without touching existing ones.
Free-text handling deliberately favors silently dropping unresolvable content over erroring — the test samples have far more "real hours + explanatory note" strings than genuinely malformed ones, so a parse failure is more often a false negative than a caught mistake. One consequence: a mistyped day inside an explicit list (e.g. "Mon and Xyz") now resolves to just the valid entries rather than erroring.

**Known unsupported cases (~7 of 472)**
Left as-is, either out of scope or genuinely ambiguous: bare 24-hour-clock times with no am/pm ("7:30-7:30"), non-clock closing times ("...to sunset"), and a couple of strings with no day information at all once the surrounding prose is stripped out.

**Testing**
154 test cases across 11 test functions (go test ./...), including a dedicated TestParseFreeformOpeningHours suite (59 cases) covering every format above plus edge cases (empty input, invalid times, invalid days, wraparound). go vet clean.

### CODE REVIEW

File: openinghours.go [hunk 1]
```diff
@@ -125,6 +126,647 @@ func ParseOpeningHours(v string) ([]OpeningHours, error) {
...
+// still required) or "11am-6pm W-F and Sun" / "10am-6pm M & W" (a day spec may itself be a
+// ","/"and"/"&"-joined list of days or day ranges). The day spec is captured permissively
+// (anything at all) because resolveDaySpec knows how to recover a valid day spec from a trailing
+// free-text note glued onto it without a proper separator (eg. "M-F daily", "daily for guests").
+var daySegmentRegexp = regexp.MustCompile(`(?i)^(\d{1,2})(?::([0-5][0-9]))?(?:\s*(am|pm))?(?:\s*-\s*|\s+to\s+)(\d{1,2})(?::([0-5][0-9]))?\s*(am|pm)[,\s]+(.+)$`)
+
+// daySpecSeparatorRegexp splits a day spec into its individual day tokens or day ranges, eg.
+// "W-F and Sun" splits into ["W-F", "Sun"], "Mon, Wed and Fri" splits into ["Mon", "Wed", "Fri"],
+// and "M & W" splits into ["M", "W"].
+var daySpecSeparatorRegexp = regexp.MustCompile(`(?i)\s*,\s*|\s+and\s+|\s*&\s*`)
```
[the remark this answers is not in the document]
Comment (bill-paren) [reply openinghours.go:263]: No.  This is not an HTML-escaped dataset, so ignoring this suggestoin

File: openinghours.go [hunk 2]
```diff
@@ -125,6 +126,647 @@ func ParseOpeningHours(v string) ([]OpeningHours, error) {
...
+		}
+	}
+
+	return ohs, nil
+}
+
+// windowSeparatorRegexp splits a segment on " and " or "&" when it's being used to join multiple
+// bare time ranges that share one trailing day spec, eg. "8am-12pm and 1pm-5pm M-F" or
+// "6:30am-6pm & 8pm-10pm daily".
+var windowSeparatorRegexp = regexp.MustCompile(`(?i)\s+and\s+|\s*&\s*`)
```
[the remark this answers is not in the document]
Comment (bill-paren) [reply openinghours.go:551]: This is not HTML-escaped data, so ignoring
